### ISSUE TITLE
The result of val dataset!

### ISSUE THREAD

**Issue by shxzhao** [Vegeta2020/SE-SSD#32]
When I eval in the val dataset, I got the result.
Evaluation official: car AP(Average Precision)@0.70, 0.70, 0.70:
bbox AP:84.92, 83.72, 83.56
bev  AP:83.67, 81.57, 75.99
3d   AP:79.95, 71.92, 71.52
aos  AP:84.66, 83.21, 82.79
car AP(Average Precision)@0.70, 0.50, 0.50:
bbox AP:84.92, 83.72, 83.56
bev  AP:85.79, 84.89, 84.54
3d   AP:85.48, 84.65, 84.30
aos  AP:84.66, 83.21, 82.79

Evaluation coco: car coco AP@0.50:0.05:0.95:
bbox AP:65.71, 63.42, 62.74
bev  AP:65.17, 62.46, 61.28
3d   AP:56.02, 52.30, 51.71
aos  AP:65.48, 63.04, 62.18

It is lower than the result of paper. Can you give me some advice?

**Comment by MuMuJun97** [comment]
@Vegeta2020  @shxzhao  I also got a lower AP results when using the model from this repository: 

```
Evaluation official_AP_11: car AP(Average Precision)@0.70, 0.70, 0.70:
bbox AP:90.61, 89.47, 89.02
bev  AP:89.94, 88.11, 87.44
3d   AP:88.44, 79.07, 78.27
aos  AP:90.51, 89.10, 88.46
car AP(Average Precision)@0.70, 0.50, 0.50:
bbox AP:90.61, 89.47, 89.02
bev  AP:90.67, 89.73, 89.38
3d   AP:90.66, 89.67, 89.28
aos  AP:90.51, 89.10, 88.46

Evaluation official_AP_40: car AP(Average Precision)@0.70, 0.70, 0.70:
bbox AP:96.27, 93.00, 90.56
bev  AP:93.25, 89.64, 87.23
3d   AP:89.73, 83.08, 80.57
aos  AP:96.12, 92.55, 89.97
car AP(Average Precision)@0.70, 0.50, 0.50:
bbox AP:96.27, 93.00, 90.56
bev  AP:96.45, 95.39, 93.02
3d   AP:96.37, 95.22, 92.89
aos  AP:96.12, 92.55, 89.97
```

**Comment by shxzhao** [comment]
Did you use your own training model or the pre training model the repository provided?

**Comment by WWW2323** [comment]
Same as you, and I found that as the training progresses, the accuracy will get lower and lower from epoch 1 to epoch 60.

**Comment by shxzhao** [comment]
I forget to set the eval result per epoch, but my result of eval dataset is so low, can you provide your set of train.py or provide the train model to me. My email is 985525622@qq.com

**Comment by WWW2323** [comment]
I didn't change the code of train.py, and my final result is same as yours.  The only thing I modified is [#14](https://github.com/Vegeta2020/SE-SSD/issues/14#issuecomment-893974439). Maybe the bad result is caused by this modification, or the author of the paper missed something. Could you give us some suggestions? @Vegeta2020

**Comment by MuMuJun97** [comment]
@shxzhao  I used the pre-trained student model the repository provided.

**Comment by pigtigger** [comment]
@WWW2323 I find the results from epoch1 to epoch60 get lower and lower too, sometimes it also will crash during epoch 50 - 60 because of "out of cuda memory ", I am wondering if the model tend to predict more proposals and get lower mAP.
Can you give me some advice?
@Vegeta2020

**Comment by FangGet** [comment]
![Selection_202](https://user-images.githubusercontent.com/11433213/129722823-ab741347-3846-4f60-a576-2c50d6ad1230.png)

This is my evaluation result using the provided model, it seems the same as in README.
If it's lower than that, I guess it's caused by data or dependencies preparation.

**Comment by WWW2323** [comment]
Maybe you need set batch size = 1 @pigtigger ,  i set the batch size = 1.
Good job! did you change any code of the repository？such as  [#14](https://github.com/Vegeta2020/SE-SSD/issues/14#issuecomment-893974439), and what's your GPU type? Thanks!!!

**Comment by WWW2323** [comment]
Maybe you need set batch size = 1 @pigtigger ,  i set the batch size = 1.
@FangGet Good job! did you change any code of the repository？such as  [#14](https://github.com/Vegeta2020/SE-SSD/issues/14#issuecomment-893974439), and what's your GPU type? Thanks!

**Comment by FangGet** [comment]
nop, I keep everything the same as the original, the model copy issue may be caused by python version, try using python 3.6.
Btw, my GPU type is RTX 2080TI

**Comment by pigtigger** [comment]
I got the same results as yours when evaluate the provided model,  but when I use it  the  as initial weights for student and teacher model and start training, the performance drops a lot  @FangGet

**Comment by gqding** [comment]
![Screenshot_20210817_213422](https://user-images.githubusercontent.com/35989403/129726385-93a30d13-5b75-4789-a853-7cd2c6e2fb0c.png)
I also obtained the same results as stated in the README!

**Comment by pigtigger** [comment]
@WWW2323 Thanks .
do you get a good results after training ? could you tell me the model you use for initialize student and teacher?  I also modified #14 ,  but not sure if it is the reason for bad performance

**Comment by WWW2323** [comment]
I have no idea, i can't even get good results before training @pigtigger

**Comment by FangGet** [comment]
@gqding I guess you should use the pre-trained model from CIA-SSD as the initial model, if SE-SSD pre-trained model is used, try to decrease the learning rate. But it's normal for performance drop as soft-target supervisor may disappear if the difference between teacher and student is ignorable, just my opinion.

**Comment by WWW2323** [comment]
I also get the same **val set** results as shown in README.md, but i run into another problem. When i use the model to inference **test set** and submit to kitti server, the results is  as follows:
![image](https://user-images.githubusercontent.com/39642019/129826998-c0a626c1-5eec-4caa-938b-a6ef345bac2d.png)


I don't know if I need to modify any configuration during testing,  because the result of SE-SSD on **test set** is this:
![image](https://user-images.githubusercontent.com/39642019/129827012-fc222a3a-0217-49fd-9879-798736b30a54.png)

If you get results similar to SE-SSD on **test set**, please @me, thanks!! Or    @Vegeta2020 can give some advice?

**Comment by shxzhao** [comment]
Do you use the pre-train model or the model you trained?@WWW2323

**Comment by WWW2323** [comment]
I use the pre-trained model provided in README to inference **val set** and **test set**. The result on the **val set** is more like the result in the **SE-SSD** paper,  but the result on the **test set** is more like the result of the **CIA-SSD**. Now I don't know if  the provided-model is the final model of SE-SSD or it is just used for SE-SSD initialization.

**Comment by shxzhao** [comment]
It should be the final model of SE-SSD.

**Comment by pigtigger** [comment]
I get the same results as yours @WWW2323 
the final model's performance is similar to CIA-SSD, so I think it is not the final model and use it to initialize SE-SSD, but the training results is even lower.

**Comment by WWW2323** [comment]
Do you mean your result on the **test set**  is the same as mine?

**Comment by pigtigger** [comment]
yes






------------------ Original ------------------
From: WWW2323 ***@***.***&gt;
Date: Wed,Aug 18,2021 0:24 PM
To: Vegeta2020/SE-SSD ***@***.***&gt;
Cc: lihe ding ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [Vegeta2020/SE-SSD] The result of val dataset! (#32)





 
Do you mean your result on the test set  is the same as mine?

**Comment by Vegeta2020** [comment]
Hi, guys. The provided model is the SE-SSD model trained on the train split and not on the trainval split, and the results given in README are based on the evaluation on the val split. (I'm afraid I cannot provide the trained model on the trainval and help you produce prediction files for submission to KITTI evaluation server, as it will violate the submission policy and lead to account banning.)
To get decent results, you must use a pre-trained model to initialize both student & teacher model. In our previous experiments, we often get decent results with a pre-trained model training from scratch, but there are only a few cases that we cannot, we guess it' related to the initialization.

**Comment by WWW2323** [comment]
Hi, everybody, do you reproduce the results? I trained a CIA-SSD model and use it for the initialization of SE-SSD. Although there is no drop as using provided model in README for initialization, there is no increase.  So sad

**Comment by pigtigger** [comment]
@WWW2323 do you mean that you use your own model trained from CIA-SSD for initialization to train SE-SSD and get almost the same results as your initialization model ?  I use CIA-SSD pretrained model in  CIA-SSD README and get a huge drop. could you send your trained CIA-SSD to me ? I want to have a try .Thanks, my email is 1059837401@qq.com

**Comment by AndyYuan96** [comment]
> Hi, everybody, do you reproduce the results? I trained a CIA-SSD model and use it for the initialization of SE-SSD. Although there is no drop as using provided model in README for initialization, there is no increase. So sad

@WWW2323 , did you use the pretrained model provided by author in CIA-SSD rep, and train using SE-SSD.

**Comment by WWW2323** [comment]
No, you need train a CIA-SSD model yourself with CIA-SSD rep. The head of CIA-SSD in CIA-SSD rep is different from the head of CIA-SSD in SE-SSD.
